appinfo/app.php

```diff
@@ -76,6 +76,7 @@ function adjustCsp(IAppContainer $container) {
 
 		foreach ($radioSources as $source) {
 			$policy->addAllowedMediaDomain($source);
+			$policy->addAllowedConnectDomain($source);
```

Comment:
This is somewhat problematic because by default, this now makes a rather large hole to the CSP. The default allowed radio sources is *:* so after this, any javascript running within ownCloud/Nextcloud is allowed to connect which ever URL it likes. This basically undermines the whole point of the Content Security Policy. Also, this is the reason, why I require each cloud administrator to whitelist the HLS radio sources they like to allow: Playing the HLS streams requires this connect-src CSP rule, and I don't want to automatically weaken the security of each and every cloud instance running this app.

Maybe the solution might be to relay all this metadata loading via our own server? So the Music app PHP backend would have a REST API to get the radio metadata and its implementation would make the calls to the icecast/shoutcast servers. I believe this would help also with the cases where the icecast/shoutcast server doesn't define the needed CORS headers.

Reply:
I agree that with php it will be easier and more secure, I was thinking that using the server to load information is more bad, if you agree I will try new method with php in the server.

Reply:
Please do. That shouldn't generate too much server load as the metadata is loaded with rather limited rate.

PS. I'll be busy with other things for the next week, so I probably can't give any more feedback during the next ~7 days. But I will review any updates as soon possible.